### ISSUE TITLE
Request for Causal AR Version Release

### ISSUE THREAD

**Issue by aengusng8** [LTH14/mar#39]
Great work! Could you please release the **_causal AR_** version used in your experiments from the paper? It would be incredibly helpful for further exploration. Thank you!

**Comment by LTH14** [comment]
Thanks for your interest! To keep this repo clean, we don't have a plan to release the causal AR code in this repo. However, it is very easy to reimplement using the current repo -- almost all hyper-parameters remain the same as MAR. The only difference is the causal attention mask and the teacher-forcing loss.

**Comment by aengusng8** [comment]
Would it be possible to share the causal-AR **_inference code_** and **_checkpoints_**? Your help would mean a lot. Many thanks in advance!

**Comment by Graduo** [comment]
> Great work! Could you please release the **_causal AR_** version used in your experiments from the paper? It would be incredibly helpful for further exploration. Thank you!

https://github.com/lucidrains/autoregressive-diffusion-pytorch?tab=readme-ov-file There is another version that lucidrains Implemented.
